api/authenticator.py

```diff
@@ -936,49 +971,90 @@ def __init__(self, library, integration, analytics=None):
                 regexp = None
         self.external_type_regular_expression = regexp
 
+        field = ConfigurationSetting.for_library_and_externalintegration(
+            _db, self.PATRON_RESTRICTION_FIELD, library, integration
+        ).value
+        if isinstance(field, basestring):
+            field = field.strip()
+        self.patron_restriction_field = field
+
+        self.patron_restriction_type = ConfigurationSetting.for_library_and_externalintegration(
+            _db, self.PATRON_RESTRICTION_TYPE, library, integration
+        ).value
+        if not self.patron_restriction_type:
+            self.patron_restriction_type = self.PATRON_RESTRICTION_TYPE_NONE
+
         restriction = ConfigurationSetting.for_library_and_externalintegration(
-            _db, self.PATRON_IDENTIFIER_RESTRICTION, library, integration
+            _db, self.PATRON_RESTRICTION_STRING, library, integration
         ).value
-        if restriction and restriction.startswith("^"):
-            # Interpret the value a regular expression
+
+        if self.patron_restriction_type == self.PATRON_RESTRICTION_TYPE_REGEX:
```

Comment:
Putting this into a helper method will make it easy to test all these possibilities without a lot of setup code.